lib/streamlit/runtime/connection_factory.py

```diff
@@ -120,6 +121,13 @@ def connection_factory(name, connection_class=None, **kwargs):
         literal as the connection_class.
       * Plugging your own ConnectionClass into st.experimental_connection.
     """
+    USE_ENV_PREFIX = "env:"
```

Comment:
how does this should correlate with streamlit-specific config options with the `STREAMLIT_` prefix?

Reply:
I don't think there are any explicit interactions we need to design for. In theory, a user could choose to have their connection name set by `env:STREAMLIT_*`, but I don't see a reason anyone would want to do this since all the config options are for unrelated things